chain/chain/src/resharding.rs

```diff
@@ -181,15 +201,49 @@ impl Chain {
         sync_hash: &CryptoHash,
         state_roots: Result<HashMap<ShardUId, StateRoot>, Error>,
     ) -> Result<(), Error> {
-        let prev_hash = *self.get_block_header(sync_hash)?.prev_hash();
+        let state_roots = state_roots?;
+        let block_header = self.get_block_header(sync_hash)?;
+        let prev_hash = block_header.prev_hash();
+
+        let child_shard_uids: Vec<_> = state_roots.keys().collect();
```

Comment:
Nit: I like the collect_vec as an alternative to collect and explicit type. It's in itertools.

Reply:
Done